### PR TITLE
feat(styles): add css-in-js-support

### PR DESCRIPTION
**Pull request by mengtzu**

### CODE REVIEW

File: config/webpack/webpack.config.js [hunk 1]
```diff
@@ -65,6 +65,38 @@ const makeCssLoaders = (options = {}) => {
...
+              'ie >= 10',
+              'Safari >= 8',
+              'iOS >= 8'
+            ]
+          })
+        ]
+      }
+    },
+    { loader: require.resolve('css-in-js-loader') },
+    { loader: require.resolve('babel-loader') }
```
Comment (mattsjones) [inline config/webpack/webpack.config.js]: It's probably a good idea to use the sku Babel config here. That way the style files will have access to the module resolution and niceties like spreads.

Comment (markdalgleish) [inline config/webpack/webpack.config.js]: Ooh, nice catch.

**Comment by mengtzu** [comment]
Added proper jsLoaders (array spread!), documentation

File: README.md [hunk 2]
```diff
@@ -50,6 +50,34 @@ export default () => (
...
+    '.exampleWrapper': {
+        ...standardWrapper,
+        fontFamily: fontFamily,
+        color: brandPrimary
+    }
+};
+```
+
+```js
+import styles from './example.css';
```
Comment (markdalgleish) [inline README.md]: Might be better to make this explicit as `./example.css.js`

File: README.md [hunk 3]
```diff
@@ -50,6 +50,34 @@ export default () => (
 );
 ```
 
+### CSS composed in Javascript (via [css-in-js-loader](https://github.com/nthtran/css-in-js-loader))
```
Comment (markdalgleish) [inline README.md]: Should we maybe rename this to `Static CSS-in-JS`?

File: README.md [hunk 4]
```diff
@@ -50,6 +50,34 @@ export default () => (
...
+
+You can import `.css.js` files into your components and use them exactly as you would a less file.  This is mostly useful when you want to take advantage of Javascript to compose styles:
```
Comment (markdalgleish) [inline README.md]: Maybe call it "a regular style sheet" instead of referencing Less?

Comment (markdalgleish) [inline README.md]: Also, total nitpick, but it's "JavaScript", not "Javascript" 🕵️

File: README.md [hunk 5]
```diff
@@ -50,6 +50,34 @@ export default () => (
...
+
+```js
+import { standardWrapper } from 'theme/wrappers';
+import { fontFamily } from 'theme/typography';
+import { brandPrimary } from 'theme/palette';
+
+export default {
+    '.exampleWrapper': {
```
Comment (markdalgleish) [inline README.md]: Should be a 2 space indent.

File: config/webpack/webpack.config.js [hunk 6]
```diff
@@ -65,6 +65,38 @@ const makeCssLoaders = (options = {}) => {
...
+              'ie >= 10',
+              'Safari >= 8',
+              'iOS >= 8'
+            ]
+          })
+        ]
+      }
+    },
+    { loader: require.resolve('css-in-js-loader') },
+    ...jsLoaders
```
Comment (markdalgleish) [inline config/webpack/webpack.config.js]: This is beautiful 👏 👏 👏 👏 👏 👏

**Comment by markdalgleish** [comment]
Probably need to discuss what this PR means for #17

**Comment by mengtzu** [comment]
Updated with @markdalgleish 's suggestions

File: config/webpack/webpack.config.js [hunk 7]
```diff
@@ -65,6 +65,38 @@ const makeCssLoaders = (options = {}) => {
...
+  return [
+    {
+      loader: require.resolve(`css-loader${server ? '/locals' : ''}`),
+      options: {
+        modules: true,
+        importLoaders: 3
+      }
+    },
+    {
+      loader: require.resolve('postcss-loader'),
```
Comment (markdalgleish) [inline config/webpack/webpack.config.js]: Sorry, looking at this again, it looks like we also need shared `cssLoaders` too?

**Comment by mengtzu** [comment]
Now with conditional, nested array spreading.  It might be 2017!

**Comment by mengtzu** [comment]
@markdalgleish probably need to ask @nicolassenechal whether there's anything left he wants in #17 that isn't covered by this + tenants.

This accommodates his primary goal of avoiding writing less, but it still doesn't provide pure css.

File: config/webpack/webpack.config.js [hunk 8]
```diff
@@ -161,6 +167,13 @@ const buildWebpackConfigs = builds.map(
               ]
             },
             {
+              test: /\.css\.js$/,
+              use: ExtractTextPlugin.extract({
+                fallback: 'style-loader',
+                use: makeCssInJsLoaders({ js: true })
```
Comment (markdalgleish) [inline config/webpack/webpack.config.js]: Shouldn't this be `makeCssLoaders` now?

Comment (mengtzu) [inline config/webpack/webpack.config.js]: Yes, fixed

**Comment by nicolassenechal** [comment]
Only thing potentially lacking is css-next, but happy to forgo it

**Pull request merged**